bolt/test/runtime/AArch64/hook-fini.test

```diff
@@ -0,0 +1,61 @@
+# Test the different ways of hooking the fini function for instrumentation (via
+# DT_FINI and via DT_FINI_ARRAY). We test the latter for both PIE and non-PIE
+# binaries because of the different ways of handling relocations (static or
+# dynamic).
+# All tests perform the following steps:
+# - Compile and link for the case to be tested
+# - Some sanity-checks on the dynamic section and relocations in the binary to
+#   verify it has the shape we want for testing:
+#   - DT_FINI or DT_FINI_ARRAY in dynamic section
+#   - No relative relocations for non-PIE
+# - Instrument
+# - Run instrumented binary
+# - Verify generated profile
+REQUIRES: system-linux,bolt-runtime
+
+RUN: %clang %p/Inputs/basic-instrumentation.s -Wl,-q -o %t.exe
+RUN: llvm-readelf -d %t.exe | FileCheck --check-prefix=DYN-FINI %s
+RUN: llvm-readelf -r %t.exe | FileCheck --check-prefix=RELOC-PIE %s
+RUN: llvm-bolt %t.exe -o %t --instrument \
+RUN:     --instrumentation-file=%t \
+RUN:     --instrumentation-file-append-pid
+RUN: rm -f %t.*.fdata
+RUN: %t
+RUN: cat %t.*.fdata | FileCheck %s
+
+RUN: %clang %p/Inputs/basic-instrumentation.s -Wl,-q,-fini=0 -o %t-no-fini.exe
```

Comment:
May I ask where did you find the possibily to disalble fini with = 0? I see it first time, is it portable solution?

Reply:
To be honest, this was a blind guess that happened to work :) It works for both ld and lld but I have no idea how portable it is. This behavior is not described in the man pages.

Do you know of a different way to achieve this?

Reply:
Well you can replace _fini symbol (I guess it is in .s file) to myfini and pass if to -fini flag if needed, otherwise it would be empty. Probably it is better option, I'm not sure if this is legit method that would continue to work..

Reply:
The .s file doesn't contain `_fini`, it comes from one of the crt files.

I just checked LLD and it adds `DT_FINI` iff `-fini` is a defined symbol. Passing any name that is not a defined symbol should disable it. There's a [test](https://github.com/llvm/llvm-project/blob/0687e4d9f310249a45c3799ec66aeeeb0efda9f7/lld/test/ELF/init-fini.s#L20-L26) that verifies this behavior so I assume it won't be changed soon.

Reply:
Interesting, thanks for the investigation!